### PR TITLE
Add dubbo-samples-cat

### PR DESCRIPTION
**Pull request by CodePrometheus**
Dubbo and [Cat](https://github.com/dianping/cat/)

### CODE REVIEW

**Comment by CodePrometheus** [comment]
@AlbumenJ  @chickenlj

File: 4-governance/dubbo-samples-cat/dubbo-samples-cat-consumer/src/test/java/org/apache/dubbo/samples/cat/consumer/CatMetricsIT.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.annotation.Resource;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class CatMetricsIT {
+
+    private static String CAT_HOST = "http://192.168.60.48:8080/cat/r/m/cat-consumer-ac171001-466309-1006?domain=cat-consumer";
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/dubbo-samples-cat-consumer/src/test/java/org/apache/dubbo/samples/cat/consumer/CatMetricsIT.java:39]:
```suggestion
    private static String CAT_HOST = "http://"+System.getProperties("cat.host","127.0.0.1")+":8080/cat/r/m/cat-consumer-ac171001-466309-1006?domain=cat-consumer";
```

File: 4-governance/dubbo-samples-cat/case-configuration.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  cat-consumer:
+    type: test
+    basedir: dubbo-samples-cat-consumer
+    tests:
+      - "**/*IT.class"
+    systemProps:
+      - zookeeper.address=cat-provider
+      - zookeeper.port=2181
+      - dubbo.address=cat-provider
+      - dubbo.port=20880
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/case-configuration.yml:35]:
```suggestion
    systemProps:
      - zookeeper.address=cat-provider
      - zookeeper.port=2181
      - dubbo.address=cat-provider
      - dubbo.port=20880
      - cat.host=cat-consumer
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 4-governance/dubbo-samples-cat/case-configuration.yml (hunk 2 above)
Comment (CodePrometheus) [reply 4-governance/dubbo-samples-cat/case-configuration.yml:35]: It seems that it can't connect to zookeeper

**Comment by AlbumenJ** [comment]
image

File: 4-governance/dubbo-samples-cat/case-configuration.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    basedir: dubbo-samples-cat-provider
+    mainClass: org.apache.dubbo.samples.cat.provider.CatProvider
+
+  cat-consumer:
+    type: test
+    basedir: dubbo-samples-cat-consumer
+    tests:
+      - "**/*IT.class"
+    systemProps:
+      - zookeeper.address=cat-provider
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/case-configuration.yml:32]:
```suggestion
      - zookeeper.address=zookeeper
```

File: 4-governance/dubbo-samples-cat/case-configuration.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# limitations under the License.
+
+services:
+  zookeeper:
+    image: zookeeper:latest
+
+  cat-provider:
+    type: app
+    basedir: dubbo-samples-cat-provider
+    mainClass: org.apache.dubbo.samples.cat.provider.CatProvider
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/case-configuration.yml:24]:
```suggestion
  cat-provider:
    type: app
    basedir: dubbo-samples-cat-provider
    mainClass: org.apache.dubbo.samples.cat.provider.CatProvider
    systemProps:
      - zookeeper.address=cat-provider
      - zookeeper.port=2181
      - dubbo.address=cat-provider
      - dubbo.port=20880
      - cat.host=cat-consumer
    waitPortsBeforeRun:
      - zookeeper:2181
    depends_on:
      - zookeeper
```

File: 4-governance/dubbo-samples-cat/case-configuration.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    tests:
+      - "**/*IT.class"
+    systemProps:
+      - zookeeper.address=cat-provider
+      - zookeeper.port=2181
+      - dubbo.address=cat-provider
+      - dubbo.port=20880
+      - cat.host=cat-consumer
+    waitPortsBeforeRun:
+      - cat-provider:2181
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/case-configuration.yml:38]:
```suggestion
      - zookeeper:2181
```

File: 4-governance/dubbo-samples-cat/case-configuration.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      - cat-provider:20880
+    depends_on:
+      - cat-provider
```
Comment (AlbumenJ) [inline 4-governance/dubbo-samples-cat/case-configuration.yml:41]:
```suggestion
    depends_on:
      - cat-provider
      - zookeeper
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by songxiaosheng** [comment]
Can you implement cat monitoring through filters in the future

**Pull request closed without merging**